### PR TITLE
Add a rule of Locale.ENGLISH to String.toUpperCase() method.

### PR DESCRIPTION
**Pull request by BruceKuiLiu**
A String is being converted to upper case, using the platform's default encoding.
This may result in improper conversions when used with international characters.
Use the String.toUpperCase( Locale l ) version should be better.
http://findbugs.sourceforge.net/bugDescriptions.html#DM_CONVERT_CASE

### CODE REVIEW

**Comment by BruceKuiLiu** [comment]
Thanks.

**Comment by garydgregory** [comment]
Shouldn't this be done with Locale.ROOT?

**Comment by BruceKuiLiu** [comment]
Right, Locale.ROOT should be the most appropriate.

File: src/main/java/org/apache/commons/lang3/time/FormatCache.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ public boolean equals(final Object obj) {
             // Eliminate the usual boilerplate because
             // this inner static class is only used in a generic ConcurrentHashMap
             // which will not compare against other Object types
-            return Arrays.equals(keys, ((MultipartKey)obj).keys);
+            return obj instanceof MultipartKey && Arrays.equals(keys, ((MultipartKey)obj).keys);
```
Comment (PascalSchumacher) [inline src/main/java/org/apache/commons/lang3/time/FormatCache.java]: Please do not include unrelated changes in a single pull request. Thanks!

**Comment by garydgregory** [comment]
I am splitting out the FastDateParser changes here: https://issues.apache.org/jira/browse/LANG-1357

**Comment by garydgregory** [comment]
I fixed some tests and code main code here and there and I think all the changes we need are in. If you feel otherwise, please update this PR based on git master. Thank you!
